stubs/pyjks/jks/bks.pyi

```diff
@@ -1,69 +1,125 @@
-from _typeshed import Incomplete
-from typing_extensions import Final, Self
+from _typeshed import SupportsKeysAndGetItem
+from typing import Any
+from typing_extensions import Final, Literal, Self, TypeAlias
 
 from .jks import TrustedCertEntry
 from .util import AbstractKeystore, AbstractKeystoreEntry
 
+_BksType: TypeAlias = Literal["bks", "uber"]
+_CertType: TypeAlias = Literal["X.509"]
+_EntryFormat: TypeAlias = Literal["PKCS8", "PKCS#8", "X.509", "X509", "RAW"]
+_BksVersion: TypeAlias = Literal[1, 2]
+
 ENTRY_TYPE_CERTIFICATE: Final = 1
 ENTRY_TYPE_KEY: Final = 2
 ENTRY_TYPE_SECRET: Final = 3
 ENTRY_TYPE_SEALED: Final = 4
+
 KEY_TYPE_PRIVATE: Final = 0
 KEY_TYPE_PUBLIC: Final = 1
 KEY_TYPE_SECRET: Final = 2
+_KeyType: TypeAlias = Literal[0, 1, 2]
 
 class AbstractBksEntry(AbstractKeystoreEntry):
-    cert_chain: Incomplete
-    def __init__(self, **kwargs) -> None: ...
+    store_type: _BksType | None
+    cert_chain: list[tuple[_CertType, bytes]]
+    def __init__(
+        self,
+        *,
+        cert_chain: list[tuple[_CertType, bytes]] = ...,
+        encrypted: bytes | None = ...,
+        store_type: _BksType = ...,
+        alias: str = ...,
+        timestamp: int = ...,
```

Comment:
As `kwargs.get("...")` is used:

```suggestion
        encrypted: bytes | None = None,
        store_type: _BksType = None,
        alias: str | None = None,
        timestamp: int | None = None,
```

The annotations for `AbstractKeystoreEntry` seem to be wrong, too.

Same for sub-classes.

Reply:
This is just sloppy API design in `pyjks`. The `__init__` constructor isn't really meant for user code (it may be useful for mocking in tests, but that's about it). Some constructable classes actually have a `new()` classmethod for this purpose, with a more reasonable interface.

Attributes like `alias`, `timestamp`, `store_type` are never `None` in real use cases, when reading or manipulating keystores.

_I could_ make them None'able in `__init__`, but changing attributes to be None'able would IMO make the stubs annoying to use and reduce its usefulness.

Reply:
So if passing `None` to these args causes unsound results, maybe instead I should remove defaults of those args? Even though the upstream code allows omitting them.

Reply:
Yes, I think removing the defaults sounds like the best idea. This will need stubtest allow entries, though.

Reply:
OK, did that. Turns out stubtest allowlists didn't need to change.